backtrace-library/src/main/cpp/backtrace-native.cpp

```diff
@@ -133,8 +133,11 @@ namespace /* anonymous */
         buffer_fd = fd;
 
         r = buffer_reader(buffer_fd, PROT_READ|PROT_WRITE);
-
+#if DEBUG_BUFFER_CREATE
+        return nullptr;
```

Comment:
why do we want to leak the `buffer_reader` when `-DDEBUG_BUFFER_CREATE`.

Also, IIUC, this `#define` enables failure injection in the `buffer_create` path? We should rename to reflect that.